templates/elasticsearch.yaml.j2

```diff
@@ -0,0 +1,124 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: elasticsearch-cluster
+  namespace: "{{ k8s_namespace }}"
+spec:
+  clusterIP: None
+  selector:
+    app: es-cluster
+  ports:
+  - name: transport
+    port: 9300
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: es-config
+  namespace: "{{ k8s_namespace }}"
+data:
+  elasticsearch.yml: |
+    cluster.name: my-elastic-cluster
+    network.host: "0.0.0.0"
+    bootstrap.memory_lock: false
+    discovery.zen.ping.unicast.hosts: elasticsearch-cluster
+    discovery.zen.minimum_master_nodes: 1
+    xpack.security.enabled: false
+    xpack.monitoring.enabled: false
+  ES_JAVA_OPTS: -Xms512m -Xmx512m
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: esnode
+  namespace: "{{ k8s_namespace }}"
+spec:
+  serviceName: elasticsearch
+  replicas: 2
+  updateStrategy:
+    type: RollingUpdate
+  selector:
+    matchLabels:
+      app: es-cluster
+  template:
+    metadata:
+      labels:
+        app: es-cluster
+    spec:
+      securityContext:
+        fsGroup: 1000
+      initContainers:
+      - name: init-sysctl
+        image: busybox
+        imagePullPolicy: IfNotPresent
+        securityContext:
+          privileged: true
+        command: ["sysctl", "-w", "vm.max_map_count=262144"]
+      containers:
+      - name: elasticsearch
+        resources:
+            requests:
+                memory: 1Gi
+        securityContext:
+          privileged: true
+          runAsUser: 1000
+          capabilities:
+            add:
+            - IPC_LOCK
+            - SYS_RESOURCE
+        image: "docker.elastic.co/elasticsearch/elasticsearch:{{ k8s_elasticsearch_version }}"
```

Comment:
(Optional) Could be nice to use the Docker Hub image for consistency with the other services

```suggestion
        image: "elasticsearch:{{ k8s_elasticsearch_version }}"
```

Reply:
Actually, nevermind. I just saw this:

<img width="1024" alt="Screen Shot 2020-01-29 at 8 39 59 PM" src="https://user-images.githubusercontent.com/160132/73412725-91382880-42d7-11ea-8e51-1928f99441d0.png">